atakama/packager.py

```diff
@@ -0,0 +1,246 @@
+        self.openssl_path = openssl or which("openssl")
+        self.openssl_path = self.openssl_path and os.path.abspath(self.openssl_path)
+        self.made_setup = False
+
+    @classmethod
+    def from_args(cls, argv):
+        """Given sys args, return a packager."""
+        args = cls.parse_args(argv)
+        return Packager(
+            src=args.src,
+            pkg=args.pkg,
+            key=args.key,
+            crt=args.crt,
+            self_signed=args.self_signed,
+            openssl=args.openssl,
+        )
+
+    @staticmethod
+    def parse_args(argv):
+        """Command line argument parser."""
+        parser = argparse.ArgumentParser(
+            description="Atakama plugin packaging helper",
+            epilog="""
+
+        An atakama plugin package consists of a python installable package, an openssl signature file,
+        and a certificate.
+
+        These three files are located in the same zip.
+
+        It is installed by installing the package into the "plugins" folder.
+
+        The python package can be:
+            - a simple zip of sources
+            - a binary wheel (which is treated, for now, as a zip anyway)
+
+        This tool simply shells out to openssl as needed to produce the signature.
+        """,
+        )
+
+        parser.add_argument("--src", help="Package source root folder.")
+        parser.add_argument("--pkg", help="Package file path (wheel, for example)")
+        parser.add_argument("--key", help="Openssl private key file", required=True)
+        parser.add_argument("--crt", help="Openssl certificate file", required=True)
+        parser.add_argument(
+            "--openssl", help="Location of openssl binary", default=which("openssl")
+        )
+        parser.add_argument(
+            "--self-signed", help="Allow a self-signed cert", action="store_true"
+        )
+        args = parser.parse_args(argv)
+        if not args.src and not args.pkg:
+            raise ValueError("Nothing to do: must specify --src or --pkg")
+        return args
+
+    def run_setup(self) -> None:
+        """Given a src directory, run the setup.py command within it."""
+        wd = os.getcwd()
+        try:
+            os.chdir(os.path.dirname(self.src))
+            print("running: ", os.getcwd(), self.setup_path, file=sys.stderr)
+            dist = run_setup(self.setup_path, script_args=["bdist_wheel"])
+            for typ, _, path in getattr(dist, "dist_files"):
+                if typ == "bdist_wheel":
+                    self.pkg = os.path.abspath(path)
+                    return
+            assert False, "Expected package not created"  # pragma: no cover
+        finally:
+            os.chdir(wd)
+
+    def has_setup(self):
+        """True if the setup.py file exists."""
+        return os.path.exists(self.setup_path)
+
+    def make_setup(self):
+        """Make a fake setup.py.  This is probably a bad idea."""
```

Comment:
yeah, can we just require it?

Reply:
maybe?  i wanted it to be easy for someone to write a simple plugin.  also, it can't actually have a requirements field.   so it's not even really anything but a zip file.  the wheel thing is just for assets. 

i could just zip it up, and not run setup.py

would actually like to discuss it.

Reply:
no out of band discussion was had.

Reply:
FWIW I tried both the "src" and "pkg" options, I like the idea of standardizing on the simpler (for us) "pkg" option and publishing documentation and a skeleton project to make things easier for others.

I can work on those things if we decide to go this route and remove the setup stuff.

Reply:
sounds good to me

Reply:
maybe a tool to inspect if a wheel was built correctly before bothering to sign anything